fullstack/src/app/api/recieve-data/route.js

```diff
@@ -41,6 +43,31 @@ export const POST = async (request) => {
             send.push(dustResults)
         }
 
+        if (humidity){
```

Comment:
This code is fine but if we want to reduce repetition make it better for scalability and flexibility i suggest somthing like this 

```
const data = {
  dust: { key: 'dust', func: DustData },
  humidity: { key: 'humidity', func: HumidityData },
  temperature: { key: 'temperature', func: TemperatureData },

};

for (const [type, { key, func }] of Object.entries(data)) {
  const value = data.uplink_message.decoded_payload[type];
  if (value !== undefined) {
    const results = await func(device_id, value);
    send.push(results);
  }
}  
```